site/frontend/src/pages/compare/shared.ts

```diff
@@ -78,3 +78,18 @@ export function getBoolOrDefault(
   }
   return defaultValue;
 }
+
+export function isBenchmarkNotFilteredByName(
+  benchmarkName: string,
+  filterName: string
+): boolean {
+  if (!filterName) return true;
```

Comment:
We should trim `filterName` at the start of the function, otherwise a space will be parsed as a regex, but it won't match anything.

Reply:
I added this early return because `filterName` is somehow nullable. Although it is not specified as such in the type, I encountered an error where `filterName.trim()` was undefined because `filterName` itself was undefined.

Reply:
Then that's probably a bug somewhere in the code that calls this function. Could you try to investigate where does this `undefined` come from?

Reply:
ahh i noticed that the project is `strict: false`, so I'll change filterName from string to string | null.

Reply:
Ah, I didn't notice that `null` is actually a valid value for the filter. Well, we could probably remove that and just use an empty string to mark a missing value, but let's keep it like it is, modify the function parameter to `string | null`, and handle the null case.